tests/Unit/DataStructures/Test_BlazeInteroperability.cpp

```diff
@@ -0,0 +1,111 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <blaze/math/CompressedMatrix.h>
+#include <blaze/math/CompressedVector.h>
+#include <blaze/math/DynamicMatrix.h>
+#include <blaze/math/DynamicVector.h>
+#include <blaze/math/StaticMatrix.h>
+#include <blaze/math/StaticVector.h>
+#include <cstddef>
+
+#include "DataStructures/CompressedMatrix/OptionCreation.hpp"
+#include "DataStructures/CompressedMatrix/Serialization.hpp"
+#include "DataStructures/CompressedVector/OptionCreation.hpp"
+#include "DataStructures/CompressedVector/Serialization.hpp"
+#include "DataStructures/DynamicMatrix/OptionCreation.hpp"
+#include "DataStructures/DynamicMatrix/Serialization.hpp"
+#include "DataStructures/DynamicVector/MakeWithValue.hpp"
+#include "DataStructures/DynamicVector/OptionCreation.hpp"
+#include "DataStructures/DynamicVector/Serialization.hpp"
+#include "DataStructures/StaticMatrix/OptionCreation.hpp"
+#include "DataStructures/StaticMatrix/Serialization.hpp"
+#include "DataStructures/StaticVector/MakeWithValue.hpp"
+#include "DataStructures/StaticVector/OptionCreation.hpp"
+#include "DataStructures/StaticVector/Serialization.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+SPECTRE_TEST_CASE("Unit.DataStructures.Blaze", "[DataStructures][Unit]") {
+  {
+    INFO("StaticVector");
+    test_serialization(blaze::StaticVector<double, 4>{0., 1., 0., 2.});
+    CHECK(make_with_value<blaze::StaticVector<double, 3>>(
+              blaze::DynamicVector<double>(3), 1.) ==
+          blaze::StaticVector<double, 3>(1.));
+    CHECK(TestHelpers::test_creation<blaze::StaticVector<double, 4>>(
+              "[0., 1., 0., 2.]") ==
+          blaze::StaticVector<double, 4>{0., 1., 0., 2.});
+  }
+  {
+    INFO("DynamicVector");
+    test_serialization(blaze::DynamicVector<double>{0., 1., 0., 2.});
+    CHECK(make_with_value<blaze::DynamicVector<double>>(
+              blaze::DynamicVector<double>(3), 1.) ==
+          blaze::DynamicVector<double>(3, 1.));
+    CHECK(TestHelpers::test_creation<blaze::DynamicVector<double>>(
+              "[0., 1., 0., 2.]") ==
+          blaze::DynamicVector<double>{0., 1., 0., 2.});
+  }
+  {
+    INFO("CompressedVector");
+    test_serialization(blaze::CompressedVector<double>{0., 1., 0., 2.});
+    CHECK(TestHelpers::test_creation<blaze::CompressedVector<double>>(
+              "[0., 1., 0., 2.]") ==
+          blaze::CompressedVector<double>{0., 1., 0., 2.});
```

Comment:
For the sparse types check that the option-created and deserialized objects are sparse.

Reply:
I believe Blaze `==` tests for mathematical equivalence, not internal-representation equivalence.  The test still passes if I change CompressedVector to DynamicVector on this line.

More generally, you don't trust Blaze to get it right in the main code.  (There are checks for zero entries, which is what I wanted a test of.)  If we can trust here Blaze we don't need those checks.  If we can't then we shouldn't be more trusting in the test.

Reply:
I added tests in a fixup, please take a look. I don't see a lower-level way of testing the internal representation.

Reply:
That's what I had in mind.  Squash.